src/promptflow-devkit/promptflow/_cli/_pf/_flow.py

```diff
@@ -522,18 +527,20 @@ def _test_flow_multi_modal(args, pf_client):
 
         pfs_port = _invoke_pf_svc()
         serve_app_port = args.port or find_available_port()
-        flow = generate_yaml_entry_without_delete(entry=args.flow)
-        # flex flow without yaml file doesn't support /eval in chat window
-        enable_internal_features = Configuration.get_instance().is_internal_features_enabled() and flow == args.flow
-        start_chat_ui_service_monitor(
-            flow=flow,
-            serve_app_port=serve_app_port,
-            pfs_port=pfs_port,
-            url_params=list_of_dict_to_dict(args.url_params),
-            init=list_of_dict_to_dict(args.init),
-            enable_internal_features=enable_internal_features,
-            skip_open_browser=args.skip_open_browser,
-        )
+        # this is a workaround given current serve app requires flow file to be under flow directory
+        # should switch to generate_yaml_entry_without_delete in the future
+        with generate_yaml_entry_under_flow_directory(entry=args.flow) as flow:
```

Comment:
not sure if we generate the yaml in temp folder like normal flow test do ?

Reply:
please check the corresponding error message. the error is caused by the limitation that current serve app doesn't accept flow_file as a parameter. I will prepare a PR for this but thought it's dangerous to do such a change on serve app near release cut

Reply:
updated the implementation